iocore/net/BIO_fastopen.cc

```diff
@@ -44,8 +57,7 @@ fastopen_destroy(BIO *bio)
   if (bio) {
     // We expect this BIO to not own the socket, so we must always
     // be in NOCLOSE mode.
-    ink_assert(bio->shutdown == BIO_NOCLOSE);
-    fastopen_create(bio);
+    ink_assert(BIO_get_shutdown(bio) == BIO_NOCLOSE);
```

Comment:
The `Data` struct is getting leaked here, but (see above) I think we can get away without it.